Adds middleware/serializers construction through Log4j configuration

ppkarwasz opened the pull request:
If the `HttpSenderMiddleware`, `EventBodySerializer` and `EventHeaderSerializer` objects require additional configuration, instantiation by class name might not be enough (see [this question on StackOverflow](https://stackoverflow.com/q/68858478/11748454)).

While maintaining backward compatibility, this PR adds the possibility to configure additional components through Log4j:

```lang-xml
<SplunkHttp>
    <SomeMiddleware attributeA="a" attributeB="b" />
    <SomeBodySerializer ... />
    <SomeHeaderSerializer ... />
</SplunkHttp>
```

The elements `SomeMiddleware`, `SomeBodySerializer` and `SomeHeaderSerializer` must be Log4j plugins of element type `middleware`, `eventBodySerializer` and `eventHeaderSerializer` respectively.

This PR contains a simple unit test.

bparmar-splunk commented:
Hi @ppkarwasz,
There are conflicts in this PR. Will you please resolve conflicts so that we can review your changes?

Thanks.

ppkarwasz commented:
@bparmar-splunk: I rebased the contribution and rewrote it to use a builder instead of a 33 parameter constructor (which I deprecated).

BTW: all `*Test` classes are excluded in your Maven Surefire configuration.
